src/Redux/Authentication/auth-reducer.ts

```diff
@@ -0,0 +1,74 @@
+import {stopSubmit} from 'redux-form';
+import {ThunkAction} from 'redux-thunk';
+import {FormAction} from 'redux-form/lib/actions';
+
+import {authAPI, ResultCodes} from '../../api';
+
+import {AppStateType} from '../redux-store';
+
+import {AuthenticationActionTypes, AuthenticationActions, AuthenticationAction} from './actions';
+
+export interface IState {
+  userId: number | null;
+  email: string | null;
+  login: string | null;
+  isAuth: boolean;
+}
+
+const initialState: IState = {
+  userId: null,
+  email: '',
+  login: '',
+  isAuth: false,
+};
+
+const authReducer = (state = initialState, action: AuthenticationAction): IState => {
+  switch (action.type) {
+    case AuthenticationActionTypes.SET_USER_DATA:
+      return {
+        ...state,
+        ...action.payload,
+      };
+
+    default:
+      return state;
+  }
+};
+
+type IThunk = ThunkAction<Promise<void>, AppStateType, unknown, AuthenticationAction>;
```

Comment:
you create this type several times.  the should be a way to do it once with a generic